### PR TITLE
Project Organization Refactoring

### LINKED ISSUE #3: Rearrange the crates
We'd like to rearrange the crates in this repo to better reflect how the other SDKs are arranged since such an arrangement is still idiomatic Rust and is more consistent with existing SDKs.

### PR DESCRIPTION
**Pull request by rylev**
Fixes #3 

This PR does the following:
* Rename the sub-crates' directories removing the azure-sdk prefix
* Collapse the various storage crates into one gated on features
* Added GitHub Actions for CI (and removed Travis)
* Update some of the READMEs and urls to point to this repo and not the old one
* Update any dependencies that could be updated without breaking things. 

Open questions:
- [ ] The storage crate is not ideally structured. I simply moved the previous crates into a single crate with submodules. We might want to rearrange things. 
- [ ] The storage crate defaults to all features. This is convenient, but means that by default the project will be slow to build. What's the right trade off here?
- [ ] Should we rename `auth_aad` to just `auth` and have AAD functionality behind a feature (so that if we add new auth varieties we can simply add new features?

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -30,8 +28,8 @@ serde-xml-rs            = "0.4"
 url                     = "2.1"
 uuid                    = { version = "0.8", features = ["v4"] }
 bytes                   = "0.5"
-hyper-rustls            = "0.20"
-failure		        = "0.1"
+hyper-rustls            = "0.21"
+failure		            = "0.1"
```
Comment (heaths) [inline core/Cargo.toml:32]: Nit: alignment issue? Or tabs vs. spaces manifested?

**Review by heaths — APPROVED** [review]
I only skimmed this. To be honest, given where we are with trying to get this in shape to really get going on new work and how much re-organization this is, I'm fine with taking this as a first stab (though nest under "sdk" if you would) and going from there. I imagine we'll have a few more iterations as we get naming figured out. So this is a great first stab and I say "go for it".

We should probably also reset versions to 1 since we'll want to rename packages anyway to align closer to our track 2 libs, where python and javascript names are probably closer to idiomatic than .NET and Java.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy: 
+      matrix: 
+        rust: [nightly]
```
[the remark this answers is not in the document]
Comment (rylev) [reply .github/workflows/build.yml:14]: Yes, there's some code that requires nightly right now. We should remove that and test against stable.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      with:
+        toolchain: ${{ matrix.rust }}
+        profile: minimal
+        override: true
+        components: rustfmt
+
+    - name: tests
+      run: cargo test --all 
+
+    - name: fmt
```
[the remark this answers is not in the document]
Comment (rylev) [reply .github/workflows/build.yml:27]: Hmm... I can look into that. I think it's possible.

**Pull request merged**